### PR TITLE
api: Allow providing decryption keys to decrypt uploaded assets

### PR DESCRIPTION
**Pull request by victorges**
**What does this pull request do? Explain your changes. (required)**
This is in order to support uploading files encrypted with Lit directly, normally stored on Bundlr.

Since the focus is on Lot, we also have an `algorithm` field which can only be AES-256-CBC for now
which is what's used for Lit. It is surprisingly cumbersome to do crypto in Go tho, so adding more 
algorithms here might not be as cheap as just calling a different function.

**Specific updates (required)**
 - Add field in task, asset and new asset payload
 - Pass the field around internally and omit it from responses
 - Make sure we delete the field in the end of a task as well
 - Improve credential cleanup logic

**How did you test each of these updates (required)**
Send lit and manually encrypted files to check it worked.

**Does this pull request close any open issues?**
Yes of a bundlr integration. Missing explicit issue.

**Checklist**

- [x] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/api/src/controllers/task.ts [hunk 1]
```diff
@@ -69,6 +69,30 @@ function taskWithIpfsUrls(
...
+        params["transcode-file"].input.url
+      );
+      result["transcode-file"].storage.url = deleteCredentials(
+        params["transcode-file"].storage.url
+      );
+      break;
+    case "upload":
+      const encryption = params.upload?.encryption;
+      if (encryption) {
+        result.upload.encryption = { ...encryption, key: "***" };
```
Comment (gioelecerati) [inline packages/api/src/controllers/task.ts:89]: I am not getting this key `***` what's stands for. Is it redacted when retrieving?

**Review by gioelecerati — APPROVED** [review]
Looking good to me

File: packages/api/src/controllers/task.ts (hunk 1 above)
Comment (victorges) [reply packages/api/src/controllers/task.ts:89]: yeah! we redact it even on the DB as well after the task is done, similarly to transcode api

**Review by victorges — COMMENTED** [review]
self-review LGTM

**Pull request merged**